### PR TITLE
add contributing doc

### PR DESCRIPTION
**Pull request by austinabell**
Opening draft for visibility, would like to get #1214 completed before this comes in to avoid having to include the version pinning semantics into this doc.

### CODE REVIEW

**Comment by austinabell** [comment]
Also noticed that the rustfmt config is nightly only so emitting warnings. Maybe it's preferred that we change the CI and the default to running `cargo +nightly fmt --all`?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+> NOTE: It is only important that you install `cargo-risczero` with a matching version of the `zkvm` crate when interacting with the proof system as a separate, pre-built process ([`ExternalProver`](https://docs.rs/risc0-zkvm/latest/risc0_zkvm/struct.ExternalProver.html)), which is currently the default. If you are using the `prove` feature on the `risc0-zkvm` crate for the host, this will compile the proving system into the host binary.
+
+#### [Git Large File Storage](https://docs.github.com/en/repositories/working-with-files/managing-large-files/installing-git-large-file-storage)
+
+Once installed, run:
+
+```console
+git lfs install
+git lfs pull
+```
```
Comment (flaub) [inline CONTRIBUTING.md:51]: This should probably be the very first step after cloning the repository.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+#### [Foundry](https://book.getfoundry.sh/getting-started/installation)
+
+```console
+curl -L https://foundry.paradigm.xyz | bash
+foundryup
+```
```
Comment (flaub) [inline CONTRIBUTING.md:58]: I think we should eventually make this part of the `xtask install` command.

File: CONTRIBUTING.md (hunk 1 above)
Comment (austinabell) [reply CONTRIBUTING.md]: I don't have the cloning step, as it's pretty intuitive for anyone contributing, but I can add it if preferred. Moved this to the start, though

**Comment by flaub** [comment]
I wonder if this is now unblocked?

**Comment by austinabell** [comment]
> I wonder if this is now unblocked?

Oops, forgot about this, sorry. Seems like this might be a helpful starting point, but I don't feel strongly about it

**Review by flaub — APPROVED** [review]

**Pull request merged**